### PR TITLE
CI plans & maven release config

### PR DESCRIPTION
**Pull request by GypsyBox**
## What:

- Release plan to make the library available in jvm project gradle build system.
- Library packages generation with the use of workflows.
- Code linting using detekt and reviewdog.

## WHY:
To improve coding standard and availability

### CODE REVIEW

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+## What:
```
Comment (mkrogulskiallegro2) [inline .github/pull_request_template.md:1]: WDYT about adding 'example' to this? So we can see what was wrong/lacked and how it is working now.

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -0,0 +1,21 @@
+name: Build libraries
+on: [push, workflow_dispatch]
+
+jobs:
+    build:
+        name: Build libraries
+        runs-on: macos-latest
```
Comment (deejay1) [inline .github/workflows/build.yml:7]: why on mac? also I'd use https://github.com/gradle/gradle-build-action for some dep caching

Comment (GypsyBox) [reply .github/workflows/build.yml:7]: Its needed to build xcframeworks for ios

File: .github/workflows/pr_review.yml [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+on: pull_request
+
+jobs:
+    detekt:
+        runs-on: macos-latest
+        steps:
+            -   name: Checkout repository
+                uses: actions/checkout@v2
+                with:
+                    fetch-depth: 0
```
Comment (mkrogulskiallegro2) [inline .github/workflows/pr_review.yml:11]: lack of space between this and next step

File: .github/workflows/pr_review.yml [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+            -   name: Setup Gradle
+                uses: gradle/gradle-build-action@v2
+
+            -   name: Set up reviewdog
+                uses: reviewdog/action-setup@v1
+
+            -   name: Run check
+                run: ./gradlew detekt -i
+
+            -   name: 'Upload report'
```
Comment (mkrogulskiallegro2) [inline .github/workflows/pr_review.yml:27]: we may consider adding 
          retention-days: ${{ env.screenshots-artifacts-retention-days }}
to not hold those for too long

File: build.gradle.kts [hunk 5]
```diff
@@ -1,9 +1,14 @@
+import io.gitlab.arturbosch.detekt.Detekt
 import org.jetbrains.kotlin.gradle.plugin.mpp.KotlinNativeTarget
 
 plugins {
     kotlin("multiplatform") version "1.6.10"
     id("maven-publish")
+    id("java-library")
+    id("signing")
     id("pl.allegro.tech.build.axion-release") version "1.13.6"
+    id("io.gitlab.arturbosch.detekt") version "1.19.0"
```
Comment (mkrogulskiallegro2) [inline build.gradle.kts:10]: maybe we should move version to another file? 

Dependencies or something,  and then use  Versions.detekt

File: build.gradle.kts [hunk 6]
```diff
@@ -24,27 +29,96 @@ kotlin {
...
+                description.set("Kotlin multiplatform JsonLogic expressions evaluation engine")
+                url.set("https://github.com/allegro/json-logic-kmp")
+                inceptionYear.set("2022")
+                licenses {
+                    license {
+                        name.set("The Apache License, Version 2.0")
+                        url.set("https://www.apache.org/licenses/LICENSE-2.0.txt")
+                    }
+                }
+                developers {
```
Comment (mkrogulskiallegro2) [inline build.gradle.kts:90]: should you be signed here specifically? Or we need more people/ more global option?

**Review by mkrogulskiallegro2 — CHANGES_REQUESTED** [review]

File: .github/workflows/pr_review.yml (hunk 4 above)
Comment (GypsyBox) [reply .github/workflows/pr_review.yml:27]: It is kept for 30 days atm based on our organization config. Do we really want to modify it?

**Review by mkrogulskiallegro2 — DISMISSED** [review]

**Review by pavlo-davydiuk — APPROVED** [review]

**Review by PaulinaSadowskaAllegro — DISMISSED** [review]

**Review by pavlo-davydiuk — APPROVED** [review]

**Review by PaulinaSadowskaAllegro — APPROVED** [review]

File: buildSrc/build.gradle.kts [hunk 7]
```diff
@@ -0,0 +1,9 @@
+plugins {
+    `kotlin-dsl`
+}
+
+repositories {
+    mavenCentral()
+    mavenLocal()
```
Comment (deejay1) [inline buildSrc/build.gradle.kts:7]: mavenLocal??

**Review by mkrogulskiallegro2 — APPROVED** [review]

File: buildSrc/build.gradle.kts (hunk 7 above)
Comment (GypsyBox) [reply buildSrc/build.gradle.kts:7]: We still use it to generate gradle dependency for the android project. Will be removed on the first release

**Review by deejay1 — APPROVED** [review]

**Pull request merged**